woocommerce/woocommerce-ios
# [Woo POS] Specialised alert view model details

Part of: #12868
Merge after #13039


## Description


This PR forms part of a series, together making the alert providers and presenters used in card payments generic.

When complete, this will mean that the POS experience can use different alert view models from the existing app's CardPresentPaymentModalViewModel, specialised for the new design, while continuing to use all the existing payments logic.

This PR introduces the new alert details view model, passing the details to the view so that we can decide how to display the messages as needed.

The happy path of connecting to a single reader is supported, with working buttons for connect, continue search, and cancel. Apologies that it's quite long... but there wasn't an obvious way of making it shorter while having a working flow.

We will initially mimic the existing view models, and we can specialise towards the new POS design after we get it working.

## Testing information


It's possible to connect a reader and take a payment, but much of the UI is not implemented yet.

Only the happy path connection views have been translated to SwiftUI, the rest will follow in the next PR.

1. Launch the app using a store eligible for Woo POS
2. Switch to POS mode
3. Tap `Connect to reader`
4. Observe that you see the reader name and connection screen, or it auto-connects if it's a known reader.
5. Tap `Connect`
6. Observe that the connection completes successfully and updates the `Reader connected` indicator.

Disconnect the reader using the Payment menu, then repeat the test while taking payment.

Test the existing app's payments and observe that they are unchanged.

https://github.com/woocommerce/woocommerce-ios/assets/2472348/b083a8ec-8451-4e85-a543-16dc89c4e537

https://github.com/woocommerce/woocommerce-ios/assets/2472348/f7c6557e-912c-49eb-a6f6-34f6fa44b5c3



---
